apps/server/src/modules/board/repo/board-do.repo.ts

```diff
@@ -0,0 +1,86 @@
+import { Utils } from '@mikro-orm/core';
+import { EntityManager } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { AnyBoardDo, BoardNode, EntityId } from '@shared/domain';
+import { BoardDoBuilder } from '@shared/domain/entity/boardnode/board-do.builder';
+import { BoardNodeBuilderImpl } from '@shared/domain/entity/boardnode/board-node-builder-impl';
+import { BoardNodeRepo } from './board-node.repo';
+
+@Injectable()
+export class BoardDoRepo {
+	constructor(private readonly em: EntityManager, private readonly boardNodeRepo: BoardNodeRepo) {}
+
+	async findById(id: EntityId): Promise<AnyBoardDo> {
```

Comment:
This returns just an `AnyBoardDo` and the caller has to narrow the type right? Can we maybe do better and query for the specific type? That would also help the om to apply the correct type constraints. Currently we are just doing a query by id and not by type.